client/src/main/java/org/apache/celeborn/client/read/LocalPartitionReader.java

```diff
@@ -61,7 +60,6 @@ public class LocalPartitionReader implements PartitionReader {
   private final FileChannel shuffleChannel;
   private List<Long> chunkOffsets;
   private int endMapIndex;
-  private AtomicBoolean hasPendingFetchTask = new AtomicBoolean(false);
```

Comment:
Deleting this will break the semantics that only one read task will be submitted to the thread pool.